File: editor/src/js/http/PageApi.ts
import {HttpService} from "../common/services/HttpService";
import {Page} from "../model/Page";
import {CommonError} from "../model/common/CommonError";

export class PageApi {
  constructor(private http: HttpService) {
  }

  pageCreatePost(): Promise<Page> {
    return this.http.POST(`/page.create`);
  }
  pageDeleteIdPost(id: string): Promise<any> {
    return this.http.POST(`/page.delete/${id}`);
  }
  pageGetGet(pageUrl: string): Promise<Page> {
    return this.http.GET(`/page.get`, { params: { pageUrl }});
  }
  pageListGet(): Promise<Page[]> {
    return this.http.GET(`/page.list`);
  }
  pageUpdatePost(pageParams: Page): Promise<Page> {
    return this.http.POST(`/page.update`, pageParams);
  }
}

File: editor-api/src/main/groovy/ish/oncourse/willow/editor/service/PageApi.java
package ish.oncourse.willow.editor.service;

import ish.oncourse.willow.editor.model.Page;
import ish.oncourse.willow.editor.model.common.CommonError;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface PageApi  {

    @POST
    @Path("/page.create")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    Page pageCreatePost();

    @POST
    @Path("/page.delete/{id}")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    void pageDeleteIdPost(@PathParam("id") String id);

    @GET
    @Path("/page.get")
    @Produces({ "application/json" })
    @AuthFilter
    Page pageGetGet(@QueryParam("pageUrl")String pageUrl);

    @GET
    @Path("/page.list")
    @Produces({ "application/json" })
    @AuthFilter
    List<Page> pageListGet();

    @POST
    @Path("/page.update")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    Page pageUpdatePost(Page pageParams);
}

